src/main/javascript/service/cat-api-service.js

```diff
@@ -323,6 +323,20 @@ function CatApiServiceProvider() {
             function $getCatApiService($http, catConversionService) {
             var catApiService = {};
 
+            /**
+             * This method allows to define (dynamic) endpoints after the configuration phase.
+             * @param {object} [settings] a new {EndpointConfig} will be created with the given settings
+             * @returns {CatApiEndpoint}
+             */
+            catApiService.$endpoint = function (settings) {
+                var name = 'dynamic-endpoint';
+                if (!_.isUndefined(settings)) {
+                    return new CatApiEndpoint(_urlPrefix, new EndpointConfig(name, settings), $http, catConversionService);
+                } else {
+                    console.error('Undefined dynamic endpoint settings');
```

Comment:
never use console directly, if you wan't to log something use the $log service

also an error should really be treated as such and not only be identified by the log level 'error' please change to

```javascript
throw new Error('Undefined dynamic endpoint settings');
```

if it can be ignored the caller should decide to do so